src/main.cpp

```diff
@@ -64,3 +70,36 @@ void setDacValue(int value) {
   mcp.setChannelValue(DACCHANNEL, value, MCP4728_VREF_INTERNAL,
                       MCP4728_GAIN_2X);
 }
+//ramps up the brightness of the light
+//also written to handle millis() overflow, I hope
+void dimmerControl(double rampTime, bool increase) {
+
+  unsigned long currTime = millis();
+  unsigned long prevTime = currTime;
+  double elapsedTime = 0;
+
+  while(elapsedTime < rampTime) {
+    currTime = millis();
+	//unsigned long should account for overflow on arithmetic
+	elapsedTime += (currTime - prevTime);
+    double ratio;
+    //dim up or down based on what is needed
+    if(increase) {
+      ratio = elapsedTime/rampTime;
+      Serial.print("Ramp up time: ");
+    }else{
+      ratio = 1 - (elapsedTime/rampTime);
+      Serial.print("Dim Down: ");
+    }
+    brightness = round(MAXBRIGHTNESS * ratio);
+    
+    //write to led, may need if statement to switch from DAC to converter pin
+    analogWrite(LED, brightness);
```

Comment:
maybe could move the portion where the actual signal is sent outside of this function, but that makes more sense when this is a non-blocking function